src/main/scala/io/github/davidgregory084/JavaMajorVersion.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022 David Gregory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.davidgregory084
+
+object JavaMajorVersion {
+  def javaMajorVersion: Int = System
+    .getProperty("java.version")
+    .stripPrefix("1.")
+    .takeWhile(_.isDigit)
+    .toInt
+}
```

Comment:
Sadly this doesn't really exist in sbt, there is a `JavaVersion` class that occupies that name in the sbt package but it doesn't offer a "major version" property of any kind.